src/core/thread/mle.cpp

```diff
@@ -1906,8 +1906,9 @@ otError Mle::SendMessage(Message &aMessage, const Ip6::Address &aDestination)
                       nonce);
 
         aesCcm.SetKey(netif.GetKeyManager().GetCurrentMleKey(), 16);
-        aesCcm.Init(16 + 16 + header.GetHeaderLength(), aMessage.GetLength() - (header.GetLength() - 1),
-                    sizeof(tag), nonce, sizeof(nonce));
+        error = aesCcm.Init(16 + 16 + header.GetHeaderLength(), aMessage.GetLength() - (header.GetLength() - 1),
+                            sizeof(tag), nonce, sizeof(nonce));
+        assert(error == OT_ERROR_NONE);
```

Comment:
MLE messages that are sent are all generated by OpenThread itself.  Currently, the only reason why `AesCcm::Init()` will fail is due to an invalid tag length.  This case should never happen in practice - if it does, we should be sure to fix the bug.  `SuccessOrExit()` is not ideal because it hides the error in practice.  Thoughts?